### DISCUSSION TITLE
colors on composited png look wrong

### DISCUSSION THREAD

**Discussion by ChameleonScales** [ImageMagick/ImageMagick#5822] [Help] [answered]
Here's a semi-transparent green circle (R=0.0, G=1.0, B=0.0, A=0.3):

![green_128](https://user-images.githubusercontent.com/19158847/204341427-05f5c889-979e-4ed6-9a0a-401a23c41a8f.png)

Here are 2 background images :
![white](https://user-images.githubusercontent.com/19158847/204516726-f8e4425e-facd-4abe-9c7d-905ddfd5e1f9.png) ![blue](https://user-images.githubusercontent.com/19158847/204516735-f4dc1f30-8d5c-4287-8e31-acb6411d079f.png)

and here's how various programs (incl. imagemagick) composite the circle on top of both backgrounds:

![grid_50pc](https://user-images.githubusercontent.com/19158847/204341844-532c5fd5-6126-43fe-9153-2a7dfd8513de.png)

The premultiplied version is only here to show that this is not an alpha premultiplication issue, since this output looks different to all the others.

For Imagemagick, I used the command:
`convert bg.png -compose atop circle.png -composite output.png`

As you can see there are 2 groups here: Imagemagick & Krita vs Gimp & Blender.

It seems to me that Blender & Gimp give a better result than Imagemagick & Krita. But why? What parameter am I missing?

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? Please supply *both* input images, so we can try your command for ourselves. Please always give these details when asking questions.

Perhaps Gimp and Blender first convert the input image, assumed to be sRGB, to linear RGB, then do the composition, then convert the result back to sRGB.

**Reply by ChameleonScales** [reply]
> Perhaps Gimp and Blender first convert the input image, assumed to be sRGB, to linear RGB, then do the composition, then convert the result back to sRGB.

That's it!
I forgot about this [great minutephysics video](https://youtu.be/LKnqECcg6Gw) on that issue.

This command outputs the same result as GIMP:
`convert -colorspace RGB bg.png -compose atop circle.png -composite -colorspace sRGB output.png`

![RGB_128](https://user-images.githubusercontent.com/19158847/204493184-7ec9e3fc-a8f2-428f-ad1f-0969d189d77e.png)

Thanks for pointing me in the right direction.
